rasa/core/train.py

```diff
@@ -26,10 +26,9 @@
     output_path: Text,
     interpreter: Optional["NaturalLanguageInterpreter"] = None,
     endpoints: "AvailableEndpoints" = None,
-    dump_stories: bool = False,
```

Comment:
I don't really like the dictionary approach in general because we are loosing the typing information and have magic strings to access the parameter. How about creating a class with types and named things? We could also do this in a separate PR? So revert the `dump_stories` part and we create an issue for the type / handling of `additional_arguments`?

Reply:
👍